includes/Modules/Analytics.php

```diff
@@ -1379,14 +1418,19 @@ protected function create_analytics_site_data_request( array $args = array() ) {
 			$request->setDateRanges( array( $date_range ) );
 		}
 
+		if ( ! empty( $args['dimension_filters'] ) ) {
+			$request->setDimensions( (array) $args['dimension_filters'] );
+		}
+
 		if ( ! empty( $args['page'] ) ) {
 			$dimension_filter = new Google_Service_AnalyticsReporting_DimensionFilter();
 			$dimension_filter->setDimensionName( 'ga:pagePath' );
 			$dimension_filter->setOperator( 'EXACT' );
 			$args['page'] = str_replace( trim( $this->context->get_reference_site_url(), '/' ), '', esc_url_raw( $args['page'] ) );
 			$dimension_filter->setExpressions( array( rawurldecode( $args['page'] ) ) );
-			$dimension_filter_clause = new Google_Service_AnalyticsReporting_DimensionFilterClause();
-			$dimension_filter_clause->setFilters( array( $dimension_filter ) );
+			$dimension_filters_clauses[] = $dimension_filter;
+			$dimension_filter_clause     = new Google_Service_AnalyticsReporting_DimensionFilterClause();
+			$dimension_filter_clause->setFilters( $dimension_filters_clauses );
 			$request->setDimensionFilterClauses( array( $dimension_filter_clause ) );
```

Comment:
@ivankruchkoff we need to move this out of the `if ( ! empty( $args['page'] ) )` condition to make sure that dimension filters work even if the page is not empty.

Reply:
Perhaps I was reading too literally into AC ```This dimension filter clause then needs to be combined with the existing dimension filter clause based on the page query parameter (see Analytics::create_analytics_site_data_request).```

Reply:
Addressed.